processor/src/main/java/org/mapstruct/ap/internal/model/BeanMappingMethod.java

```diff
@@ -608,12 +612,11 @@ private void reportErrorForUnmappedTargetPropertiesIfRequired() {
             // fetch settings from element to implement
             ReportingPolicyPrism unmappedTargetPolicy = getUnmappedTargetPolicy();
 
-            //we handle forged methods differently than the usual source ones. in
-            if ( method instanceof ForgedMethod ) {
+            //we handle automapping forged methods differently than the usual source ones. in
+            if ( method instanceof ForgedMethod && ( (ForgedMethod) method ).isAutoMapping() ) {
```

Comment:
With my latest change it becomes more apparent: 
* the case here is forged bean mapping method
* more-over, this executed only when its a 'name-based' mapping. 

So what if we implement a method on method called: `isForgedNameBased`?